core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/repository/BaseRepositoryPreCommitHook.java

```diff
@@ -42,8 +43,18 @@ public BaseRepositoryPreCommitHook(Logger log) {
 	
 	@Override
 	public void run(StagingArea staging) {
+		boolean needsDocumentUpdate;
+		if (staging.getContext() instanceof RepositoryTransactionContext) {
+			RepositoryTransactionContext transactionContext = (RepositoryTransactionContext) staging.getContext();
```

Comment:
In Java 17 you can combine `instanceof` checks with variable declarations like so:

```suggestion
		if (staging.getContext() instanceof RepositoryTransactionContext transactionContext) {
```